### PR TITLE
Use log directory that we will be able to write to

### PR DESCRIPTION
**Pull request by MadLittleMods**
Use log directory that we will be able to write to.

Fix https://github.com/matrix-org/dendrite/issues/1644

### Pull Request Checklist

* [ ] I have added any new tests that need to pass to `sytest-whitelist` as specified in [docs/sytest.md](https://github.com/matrix-org/dendrite/blob/master/docs/sytest.md)
* [ ] ~~Pull request includes a [sign off](https://github.com/matrix-org/dendrite/blob/master/docs/CONTRIBUTING.md#sign-off)~~

Signed-off-by: `Your Name <your@email.example.org>`

### CODE REVIEW

File: dendrite-config.yaml [hunk 1]
```diff
@@ -383,4 +383,4 @@ logging:
 - type: file
   level: info
   params:
-    path: /var/log/dendrite
+    path: ./logs
```
Comment (MadLittleMods) [inline dendrite-config.yaml:386]: Ran this locally a few time and it creates the `./logs` directory just fine.

File: .gitignore [hunk 2]
```diff
@@ -19,6 +19,7 @@
 /_test
```
Comment (MadLittleMods) [inline .gitignore:19]: It's unclear to me why the Sytest job is failing and what it is complaining about, https://buildkite.com/matrix-dot-org/dendrite/builds/4671#46eca488-3a58-4493-8b43-9866ec5e9760

**Review by Kegsay — CHANGES_REQUESTED** [review]
Thanks for this! We use this path in numerous other places though, I think I need @neilalexander to help on which need to be changed (the ones for docker are probably fine staying as `/var/log`:
```
/cmd/generate-config/main.go:103:				"path": "/var/log/dendrite",
/build/docker/config/dendrite-config.yaml:344:    path: /var/log/dendrite
```

File: .gitignore (hunk 2 above)
Comment (Kegsay) [reply .gitignore:19]: The key change test is flakey annoyingly.

**Comment by neilalexander** [comment]
... which makes me realise that we should probably update the Docker Compose sample files to map the media and log directories somewhere outside of the container.

**Comment by MadLittleMods** [comment]
@Kegsay It looks like both of those places are related to the Docker images. `generate-config` is only used in Docker and isn't documented elsewhere. And the other instance `/build/docker/config/dendrite-config.yaml` is self-explanatory as Docker.

Anything else stopping this from merging? Seems like we want to iterate a bit on the Docker setup but we can handle in another PR.

**Comment by Kegsay** [comment]
Agreed.

**Pull request merged**